hr_payslip_move_date/models/hr_payslip.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+# © 2014 Eficent
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import fields, models, api
+
+
+class HrPayslip(models.Model):
+    _inherit = 'hr.payslip'
+
+    move_date = fields.Date(
+        string='Force move date',
+        required=False,
+    )
+
+    @api.onchange('move_date')
+    def onchange_move_date(self):
+        period_obj = self.env['account.period']
+        if self.move_date:
+            period_ids = period_obj.find(dt=self.move_date)
+            if period_ids:
+                self.period_id = period_ids[0]
```

Comment:
Please add a check if there are periods defined for that date, if it's a new fiscalyear, in January, probably the new fiscalyear is not opened...